src/main/kotlin/org/rust/lang/core/lexer/RustStringLiteralLexer.kt

```diff
@@ -0,0 +1,137 @@
+package org.rust.lang.core.lexer
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.StringEscapesTokenTypes.*
+import com.intellij.psi.tree.IElementType
+
+private const val BYTE_ESCAPE_LENGTH = "\\x00".length
+private const val UNICODE_ESCAPE_MIN_LENGTH = "\\u{0}".length
+private const val UNICODE_ESCAPE_MAX_LENGTH = "\\u{000000}".length
+
+// TODO: Maybe implement lexing of prefixes and suffixes?
```

Comment:
Yeah, they are uniquely identified. My point is different. After some refactoring, this code might be very useful for *extracting* both prefixes and suffixes, not *identifying* them. Providing detailed information about all characteristic parts (prefixes, suffixes, quotes, escapes, maybe "escapable" characters) of a string literal, this lexer would be quite powerful base for a wide range of annotators and inspections, which otherwise would have to do all this analysis themselves, probably using some extension methods on string.